### PR TITLE
Add group parameter support to win_frewall_rule

### PR DESCRIPTION
**Pull request by ghost**
##### SUMMARY

Changes in this PR will add support to specify `group` parameter for the firewall rule.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
`win_firewall_rule`

##### ADDITIONAL INFORMATION

Current module doesn't support `group` parameter for `win_firewall_rule` module. This is a limitation for the user who needs to specify the group of the Firewall role to set/create.

Before:
```
  - name: test
    win_firewall_rule:
      name: 'Test Rule'
      program: 'c:\app\app1.exe'
      direction: In
      group:  TestGrp
      action: Allow
      state: Present
```
Result: Rule created without setting the group.

After:
 ``` 
- name: test
    win_firewall_rule:
      name: 'Test Rule'
      program: 'c:\app\app1.exe'
      direction: In
      group:  TestGrp
      action: Allow
      state: Present
```
Result: Rule created by setting the group.

### CODE REVIEW

**Comment by kvprasoon** [comment]
@jborean93 Will you be able to do a quick review ?

File: lib/ansible/modules/windows/win_firewall_rule.ps1 [hunk 1]
```diff
@@ -170,6 +172,7 @@ $name = Get-AnsibleParam -obj $params -name "name" -failifempty $true
 $description = Get-AnsibleParam -obj $params -name "description" -type "str"
 $direction = Get-AnsibleParam -obj $params -name "direction" -type "str" -failifempty $true -validateset "in","out"
 $action = Get-AnsibleParam -obj $params -name "action" -type "str" -failifempty $true -validateset "allow","block"
+$group = Get-AnsibleParam -obj $params -name "group" -type "str" -failifempty $false
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_firewall_rule.ps1]: `-failifempty $false` is the default and not needed.

**Review by jborean93 — CHANGES_REQUESTED** [review]
You will also have to add the documentation, looking at the MS docs for [Grouping](https://docs.microsoft.com/en-us/windows/desktop/api/netfw/nf-netfw-inetfwrule-get_grouping) it seems like it should be in a specific form. We should document this but it would be even better to hide this layer and just specify a group name normally if possible.

Lastly we would want to see some integration tests at https://github.com/ansible/ansible/tree/devel/test/integration/targets/win_firewall_rule. Feel free to reach out if you need help running the tests. https://docs.ansible.com/ansible/devel/dev_guide/testing_integration.html#windows-tests has a brief guide on how to do this.

**Comment by prasoon-philips** [comment]
@jborean93 Need you help with the errors please.

**Comment by jborean93** [comment]
@prasoon-philips you need to rebase your commit per the link from ansibot here https://github.com/ansible/ansible/pull/54289#issuecomment-478323430. You also used Windows line endings `\r\n` when we expect the linux line endings `\n` hence why the diff shows a completely new file and sanity checks have failed. Lastly you need to add `version_added: '2.8'` to the new `group` option in the docs so people know this option was added in Ansible 2.8.

**Comment by prasoon-philips** [comment]
Thanks @jborean93 for the tips, I'm closing this PR and this will be tracked via my original account through [this](https://github.com/ansible/ansible/pull/55109) PR, this accnt will be deleted soon.

**Pull request closed without merging**